tests/components/anglian_water/conftest.py

```diff
@@ -38,6 +38,14 @@ def mock_smart_meter() -> SmartMeter:
     mock.latest_read = 50
     mock.yesterday_water_cost = 0.5
     mock.yesterday_sewerage_cost = 0.5
+    # For diagnostic tests
+    mock.to_dict.return_value = {
+        "serial_number": mock.serial_number,
+        "latest_read": mock.latest_read,
+        "get_yesterday_consumption": mock.get_yesterday_consumption,
+        "yesterday_water_cost": mock.yesterday_water_cost,
+        "yesterday_sewerage_cost": mock.yesterday_sewerage_cost,
+    }
```

Comment:
This doesn't match the library implementation
https://github.com/pantherale0/pyanglianwater/blob/f5f1e608361a0544f462c43b3249b693ba676d5f/pyanglianwater/meter.py#L70-L77